### ISSUE TITLE
Search problem with IPTorrents

### ISSUE THREAD

**Issue by berong91** [CouchPotato/CouchPotatoServer#6625]
### Steps to reproduce:
1. Set torrents in search option
2. Set IpTorrents as my only search provider
3. Try to search for a movie

### Information:
Movie(s) I have this with: Hellraiser
Quality of the movie being searched: Best (BR-Disk, 1080p, 720p, BD-Rip)
Providers I use: IpTorrents
Version of CouchPotato: git:(CouchPotato:CouchPotatoServer master) 13a19423 (2016-05-29 11:30:23)
Running on: July 18, 5:20PM PST

### Logs:
```
07-19 02:18:09 INFO [ie.providers.info.omdbapi] Found: Hellraiser (1987) 
07-19 02:18:09 INFO [core.media.movie.searcher] Search for Hellraiser in BR-Disk ignoring ETA 
07-19 02:18:10 INFO [core.media.movie.searcher] Search for Hellraiser in 1080p ignoring ETA 
07-19 02:18:10 INFO [core.media.movie.searcher] Search for Hellraiser in 720p ignoring ETA 
07-19 02:18:10 INFO [core.media.movie.searcher] Search for Hellraiser in BR-Rip ignoring ETA 
07-19 02:18:10 ERROR [   couchpotato.core.event] Error in event "quality.single", that wasn't caught: Traceback (most recent call last):
  File "/home14/berong91/software/CouchPotatoServer/couchpotato/core/event.py", line 15, in runHandler
    return handler(*args, **kwargs)
  File "/home14/berong91/software/CouchPotatoServer/couchpotato/core/plugins/quality/main.py", line 117, in single
    quality = db.get('quality', identifier, with_doc = True)['doc']
  File "/home14/berong91/software/CouchPotatoServer/libs/CodernityDB/database_super_thread_safe.py", line 43, in _inner
    res = f(*args, **kwargs)
  File "/home14/berong91/software/CouchPotatoServer/libs/CodernityDB/database.py", line 944, in get
    raise RecordNotFound("Not found")
RecordNotFound: Not found
encoding=UTF-8 debug=False args=['--daemon'] app_dir=/home14/berong91/software/CouchPotatoServer data_dir=/home14/berong91/.couchpotato desktop=None options=Namespace(config_file='/home14/berong91/.couchpotato/settings.conf', console_log=False, daemon=True, data_dir=None, debug=False, pid_file='/home14/berong91/.couchpotato/couchpotato.pid', quiet=False)  
07-19 02:18:10 ERROR [   couchpotato.core.event] Error in event "movie.searcher.single", that wasn't caught: Traceback (most recent call last):
  File "/home14/berong91/software/CouchPotatoServer/couchpotato/core/event.py", line 15, in runHandler
    return handler(*args, **kwargs)
  File "/home14/berong91/software/CouchPotatoServer/couchpotato/core/media/movie/searcher.py", line 200, in single
    log.info('Search for %s in %s%s', (default_title, quality['label'], ' ignoring ETA' if always_search or ignore_eta else ''))
TypeError: 'NoneType' object has no attribute '__getitem__'
encoding=UTF-8 debug=False args=['--daemon'] app_dir=/home14/berong91/software/CouchPotatoServer data_dir=/home14/berong91/.couchpotato desktop=None options=Namespace(config_file='/home14/berong91/.couchpotato/settings.conf', console_log=False, daemon=True, data_dir=None, debug=False, pid_file='/home14/berong91/.couchpotato/couchpotato.pid', quiet=False)  
```

**Comment by CouchPotatoBot** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Fill in all the dots. They're there for a reason
- [x] Post logs, either inline (for smaller logs) or using [Pastebin](http://pastebin.com/)
- [x] Please wrap inline logs inside ``` for readability
- [ ] Enabled debug logging in Settings > Advanced (restart CP, be sure to disable after the bug is fixed)
- [ ] Post full version information

Please make sure you also read [contribution guide](https://github.com/RuudBurger/CouchPotatoServer/blob/develop/contributing.md#issues) and followed all the steps. 
Make the title describe your issue. Having "CP not working" or "I get this bug" for 100 issues, isn't really helpful. My master will close issues if there isn't enough information. On a good day he will tag the issue on close with the reason (like `can't reproduce`), but usually he won't, the lazy asshat.

Sometimes my master seems like a grumpy cat and responds with short answers. This isn't (always) because he hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let him know. Maybe he can teach me his awesome ways.

Thanks!

**Comment by Prime-Omega** [comment]
Same issue here:

```
07-19 13:27:28 INFO
[ie.providers.info.omdbapi] Found: Batman v Superman: Dawn of Justice (2016)
07-19 13:27:29 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/eta/tt2975590/, data: []
07-19 13:27:29 INFO
[core.media.movie.searcher] Search for Batman v Superman: Dawn of Justice in 720p ignoring ETA
07-19 13:27:29 INFO
[hpotato.core.plugins.base] Opening url: post https://iptorrents.eu/, data: ['username', 'login', 'password']
07-19 13:27:30 INFO
[hpotato.core.plugins.base] Opening url: get https://iptorrents.eu/t?l48=&q=%22batman+v+superman+dawn+of+justice%22+2016&qf=#torrents&p=1, data: []
07-19 13:27:31 INFO
[edia._base.providers.base] Found: score(366) on IPTorrents: Batman v Superman Dawn of Justice (2016) 720Pad BluRay AAC x264-aSOUL@BluRG
07-19 13:27:32 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW4gZGF3biBvZiBqdXN0aWNlIDIwMTYgcGxkdWIgZHVhbCA3MjBwIGJsdXJheSB4MjY0LWZsYW1l/, data: []
07-19 13:27:32 INFO
[edia._base.providers.base] Found: score(254) on IPTorrents: Batman v Superman Dawn of Justice 2016 PLDUB DUAL 720p BluRay x264-FLAME
07-19 13:27:33 INFO
[edia._base.providers.base] Found: score(943) on IPTorrents: Batman v Superman Dawn of Justice 2016 EXTENDED 720p BluRay H264 AAC PapaFatHead mkv
07-19 13:27:36 INFO
[edia._base.providers.base] Found: score(1187) on IPTorrents: Batman v Superman Dawn of Justice 2016 Extended 720p BluRay x264 x0r
07-19 13:27:36 INFO
[hpotato.core.plugins.base] Opening url: get https://iptorrents.eu/t?l48=&q=%22batman+v+superman+dawn+of+justice%22+2016&qf=#torrents&p=1, data: []
07-19 13:27:39 INFO
[core.media.movie.searcher] Search for Batman v Superman: Dawn of Justice in BR-Rip ignoring ETA
07-19 13:27:39 INFO
[hpotato.core.plugins.base] Opening url: get https://iptorrents.eu/t?l7=&l48=&l20=&q=%22batman+v+superman+dawn+of+justice%22+2016&qf=#torrents&p=1, data: []
07-19 13:27:40 INFO
[edia._base.providers.base] Found: score(375) on IPTorrents: Batman v Superman Dawn of Justice (2016) 720Pad BluRay AAC x264-aSOUL@BluRG
07-19 13:27:40 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW4gZGF3biBvZiBqdXN0aWNlIDIwMTYgZXh0ZW5kZWQgbXVsdGlzdWJzIGJkcmlwIDEwODBwIGRkLTUgMSB4MjY1LWRyc2k=/, data: []
07-19 13:27:40 INFO
[edia._base.providers.base] Found: score(802) on IPTorrents: Batman v Superman Dawn of Justice 2016 Extended MultiSubs BDRip 1080p DD-5 1 x265-DrSi
07-19 13:27:40 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW4gZGF3biBvZiBqdXN0aWNlIDIwMTYgcGxkdWIgZHVhbCA0ODBwIHgyNjQtbXNk/, data: []
07-19 13:27:40 INFO
[edia._base.providers.base] Found: score(377) on IPTorrents: Batman v Superman Dawn of Justice 2016 PLDUB DUAL 480p x264-mSD
07-19 13:27:40 INFO
[edia._base.providers.base] Found: score(261) on IPTorrents: Batman v Superman Dawn of Justice 2016 PLDUB DUAL 720p BluRay x264-FLAME
07-19 13:27:40 INFO
[edia._base.providers.base] Found: score(972) on IPTorrents: Batman v Superman Dawn of Justice 2016 EXTENDED 720p BluRay H264 AAC PapaFatHead mkv
07-19 13:27:41 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW4gZGF3biBvZiBqdXN0aWNlIDIwMTYgZXh0ZW5kZWQgYnJyaXAgbXVsdGlzdWIgYWMzIHgyNjQtYWNt/, data: []
07-19 13:27:41 INFO
[edia._base.providers.base] Found: score(755) on IPTorrents: Batman v Superman Dawn of Justice 2016 Extended BRRip MultiSub AC3 x264-ACM
07-19 13:27:41 INFO
[edia._base.providers.base] Found: score(268) on IPTorrents: Batman v Superman Dawn of Justice 2016 BD-Rip 1080p x265 ac3 6ch aac 2ch -Dtech mkv
07-19 13:27:41 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW46IGRhd24gb2YganVzdGljZSAyMDE2IGV4dGVuZGVkIGN1dCA3MjBwIGJkcmlwIGRkNSAxIHgyNjQtaXRzYXQ=/, data: []
07-19 13:27:41 INFO
[edia._base.providers.base] Found: score(538) on IPTorrents: Batman v Superman: Dawn of Justice 2016 Extended Cut 720p BDRip DD5 1 x264-ITSat
07-19 13:27:42 INFO
[hpotato.core.plugins.base] Opening url: get https://api.couchpota.to/validate/YmF0bWFuIHYgc3VwZXJtYW4gZGF3biBvZiBqdXN0aWNlIDIwMTYgZXh0ZW5kZWQgMTA4MHAgYmx1cmF5IHgyNjUgaGV2YyA2Y2gtbXJu/, data: []
07-19 13:27:42 INFO
[edia._base.providers.base] Found: score(1421) on IPTorrents: Batman v Superman Dawn of Justice 2016 EXTENDED 1080p BluRay x265 HEVC 6CH-MRN
07-19 13:27:42 INFO
[edia._base.providers.base] Found: score(1216) on IPTorrents: Batman v Superman Dawn of Justice 2016 Extended 720p BluRay x264 x0r
07-19 13:27:44 ERROR
[ couchpotato.core.event] Error in event "quality.single", that wasn't caught: Traceback (most recent call last): File "/volume1/@appstore/couchpotatoserver/share/CouchPotatoServer/couchpotato/core/event.py", line 15, in runHandler return handler(*args, **kwargs) File "/volume1/@appstore/couchpotatoserver/share/CouchPotatoServer/couchpotato/core/plugins/quality/main.py", line 117, in single quality = db.get('quality', identifier, with_doc = True)['doc'] File "/usr/local/couchpotatoserver/share/CouchPotatoServer/libs/CodernityDB/database_super_thread_safe.py", line 43, in _inner res = f(*args, **kwargs) File "/usr/local/couchpotatoserver/share/CouchPotatoServer/libs/CodernityDB/database.py", line 944, in get raise RecordNotFound("Not found") RecordNotFound: Not found encoding=UTF-8 debug=False args=['--daemon', '--pid_file', '/usr/local/couchpotatoserver/var/couchpotatoserver.pid', '--config_file', '/usr/local/couchpotatoserver/var/settings.conf'] app_dir=/usr/local/couchpotatoserver/share/CouchPotatoServer data_dir=/usr/local/couchpotatoserver/var desktop=None options=Namespace(config_file='/usr/local/couchpotatoserver/var/settings.conf', console_log=False, daemon=True, data_dir=None, debug=False, pid_file='/usr/local/couchpotatoserver/var/couchpotatoserver.pid', quiet=False)
07-19 13:27:44 ERROR
[ couchpotato.core.event] Error in event "movie.searcher.single", that wasn't caught: Traceback (most recent call last): File "/volume1/@appstore/couchpotatoserver/share/CouchPotatoServer/couchpotato/core/event.py", line 15, in runHandler return handler(*args, **kwargs) File "/volume1/@appstore/couchpotatoserver/share/CouchPotatoServer/couchpotato/core/media/movie/searcher.py", line 200, in single log.info('Search for %s in %s%s', (default_title, quality['label'], ' ignoring ETA' if always_search or ignore_eta else '')) TypeError: 'NoneType' object has no attribute '__getitem__' encoding=UTF-8 debug=False args=['--daemon', '--pid_file', '/usr/local/couchpotatoserver/var/couchpotatoserver.pid', '--config_file', '/usr/local/couchpotatoserver/var/settings.conf'] app_dir=/usr/local/couchpotatoserver/share/CouchPotatoServer data_dir=/usr/local/couchpotatoserver/var desktop=None options=Namespace(config_file='/usr/local/couchpotatoserver/var/settings.conf', console_log=False, daemon=True, data_dir=None, debug=False, pid_file='/usr/local/couchpotatoserver/var/couchpotatoserver.pid', quiet=False)
```

**Comment by Prime-Omega** [comment]
I did some more research on this and fixed the issue for myself. Turns out it has nothing to do with IPTorrents but actually with your quality settings. (Kind of obvious from the Error in event "quality.single)

Anyway, go to your quality settings and make sure the last quality you select is always +any quality.
It is a known bug which makes CP go haywire...

**Comment by berong91** [comment]
I'm not quite sure about +Any quality. Is that a custom quality profile or something else? Could you take a picture of your setting that solved the problem?

**Comment by berong91** [comment]
I couldn't find "Any quality" anywhere, as this is the list of choice I have right now: [Picture](https://puu.sh/qawjl/61cb7b7488.png).

Cause the CoachPotato is hosted by my seedbox service right now, I wonder whether the server is outdated. It's Version13a19423 right now.

**Comment by Prime-Omega** [comment]
Was still asleep when I typed that comment, I meant +Add another quality.
Make sure that is the last option in the profile you are using and then try again.

If you refresh the option might disappear again but that's okay, just give it a go.

**Comment by berong91** [comment]
The quick fix clears the "quality.single" error, completely, but for some reason, CoachPotato can't detect any error from IpTorrents return results. Is that happen to you?

````
07-22 17:21:42 DEBUG [o.core.notifications.core] Notifying frontend 
07-22 17:21:42 DEBUG [o.core.notifications.core] Done notifying frontend 
07-22 17:21:42 DEBUG [couchpotato.core.settings] Property "last_ignored_eta.f8a1d42068d74ab182c0295075814fcc" doesn't exist: Traceback (most recent call last):
RecordNotFound: Not found
 
07-22 17:21:42 INFO [core.media.movie.searcher] Search for Inception in 1080p ignoring ETA 
07-22 17:21:42 INFO [hpotato.core.plugins.base] Opening url: get https://iptorrents.eu/t?l48=&q=%22inception%22+2010&qf=#torrents&p=1, data: [] 
07-22 17:21:42 DEBUG [hpotato.core.plugins.base] Setting cache 85bf0b645fb5a11f7604017f478a7dfe 
07-22 17:21:42 DEBUG [core.media.movie.searcher] Nothing found for Inception in 1080p 
07-22 17:21:42 INFO [core.media.movie.searcher] Search for Inception in 720p ignoring ETA 
07-22 17:21:42 DEBUG [hpotato.core.plugins.base] Getting cache 85bf0b645fb5a11f7604017f478a7dfe 
07-22 17:21:43 DEBUG [core.media.movie.searcher] Nothing found for Inception in 720p 
07-22 17:21:43 DEBUG [o.core.notifications.core] Notifying frontend 
07-22 17:21:43 DEBUG [o.core.notifications.core] Done notifying frontend 
07-22 17:21:43 DEBUG [o.core.notifications.core] Notifying frontend 
07-22 17:21:43 DEBUG [o.core.notifications.core] Done notifying frontend 
07-22 17:21:43 DEBUG [o.core.notifications.core] Getting messages with id: cf2c212f-2211-4665-8bd2-a3973da064ec 
07-22 17:21:43 DEBUG [o.core.notifications.core] Returning for cf2c212f-2211-4665-8bd2-a3973da064ec 3 messages 
````

**Comment by Prime-Omega** [comment]
Clear your IPTorrents configuration completely and disable it in the searcher options. After that reboot CP and try entering them again.

Also golden tip, do you have any special characters in your IPTorrents password? Try changing your password to something with only letters/numbers like "adv5rylli2rf5".

**Comment by Prime-Omega** [comment]
Managed to get it fixed?

**Comment by berong91** [comment]
Hi again.
So I did more research and found out that for IpTorrents, the server also needs to install 'python-XML' library as well. After a talk with tech support guy, we managed to install that lib and that resolved everything. 

It is specified in the installing script of CouchPotato, but it seems like they totally missed it.

I guess I'll close this thread for now. Thank you very much for your help and have a nice day.
